client/src/commands/run.ts

```diff
@@ -26,54 +28,62 @@ function getCode(outputHtml: boolean, selected = false): string {
 }
 
 async function _run(selected = false) {
-  const outputHtml = !!workspace
-    .getConfiguration("SAS")
-    .get("session.outputHtml");
-  const code = getCode(outputHtml, selected);
+  if (loading) return;
+  loading = true;
 
-  await window.withProgress(
-    {
-      location: ProgressLocation.Notification,
-      title: "Connecting to SAS session...",
-    },
-    setup
-  );
+  try {
```

Comment:
I'm not a fan of huge try blocks. How about adding another function handling _run.catch().finally()?

Reply:
all right, I'll revise this.